metaseq/trainer.py

```diff
@@ -1033,13 +1033,16 @@ def _prepare_sample(self, sample, is_dummy=False):
         if self.cuda:
             sample = utils.move_to_cuda(sample)
 
-        def apply_half(t):
-            if t.dtype is torch.float32:
+        def apply_precision_conversion(t):
```

Comment:
The end result of this function and `floating_point_precision_convertor` seem to be the same. What's the reason for two methods here? 

Also it seems like goal is to always halve / lower the precision. Another naming nit: add "lower" or "half" to the name (going back to apply_half lol)

Reply:
naming is hard. I would bias towards something like `lower_precision` or `apply_16bit`

I'm happy with the logic changes (we don't ignore the config in the case of fp16 in).